### PR TITLE
Added CancelTaskException class

### PR DESCRIPTION
**Pull request by JavierLeon9966**
## Introduction

Adds CancelTaskException class which cancels the current task by throwing this exception.
### Relevant issues

None
### API changes

Added CancelTaskException class, by throwing this exception the current task gets canceled.
### Behavioural changes

None
## Backwards compatibility

None
## Follow-up

None
## Tests

711a05c

### CODE REVIEW

**Review by SOF3 — APPROVED** [review]
Looks good, with reasonable use case.

File: src/scheduler/CancelTaskException.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\scheduler;
+
+final class CancelTaskException extends \Exception{
```
Comment (dktapps) [inline src/scheduler/CancelTaskException.php:26]: Please add a doc comment with a brief explanation of what this exception is used for.

File: tests/phpunit/scheduler/TaskSchedulerTest.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	public function testCancel() : void{
+		$task = $this->scheduler->scheduleTask(new CancelTask());
+		$cancelled = false;
+		try{
+			$task->run();
+		}catch(CancelTaskException $e){
+			$cancelled = true;
+		}
+		self::assertTrue($cancelled, "Task was not cancelled");
```
Comment (dktapps) [inline tests/phpunit/scheduler/TaskSchedulerTest.php:49]: You should call `this->scheduler->mainThreadHeartbeat()` in here instead, and then assert that `task->isCancelled()` is true.

**Review by dktapps — COMMENTED** [review]
Looks good, minus one or two minor nits.

**Comment by dktapps** [comment]
`TaskScheduler->run()` and `Task->onRun()` should have `@throws CancelTaskException` annotations added.

**Comment by JavierLeon9966** [comment]
Ok, ready

**Pull request merged**